src/main/snapshot.ts

```diff
@@ -57,7 +58,10 @@ export const getSnapshotDownloadTarget = async (
   storePath: string,
   basePath: string,
   userDataPath: string,
-  token: CancellationToken
+  token: CancellationToken,
+  mixpanel: Mixpanel | null,
+  mixpanelUUID: string | null,
+  ip: string | null
```

Comment:
1. How about keeping `cancellationToken` as last argument? (it isn't TS's convention but C#'s 😅 )
2. If we need 3 arguments, in other words, it seems that we need to create a separate data type that includes these args. 🤔